### PR TITLE
Switched the CheckboxInput component to use Checkbox.Item to stay mor…

### PR DESCRIPTION
**Pull request by himrocks33**
Switched the CheckboxInput component to use Checkbox.Item to stay more consistent with RN paper and allow for the mode parameter.

This will use Text from RN Paper for the label as well keeping the theme consistent.

### CODE REVIEW

**Review by Thodor12 — CHANGES_REQUESTED** [review]
React native paper uses checkboxes after the line of text, that doesn't look good in the select.
I would not accept the PR in the current state without providing an alternative to still have the text default to being right aligned.

**Comment by himrocks33** [comment]
@Thodor12 Please note in this PR I have set the 'position' property to 'leading'.  This puts the checkbox on the left side of the text.  The implementation is the same as before.  It just stays more consistent with React Native Paper's styling.  It also adds the mode property.  Thanks for the review.

**Comment by himrocks33** [comment]
Now also includes bug fixes for #53 and #49 as per @srivastavaanurag79 request.

@CarlosSLoureiro or @Thodor12  could you please review?

**Comment by Thodor12** [comment]
With all the individual property additions, isn't it easier to just inherit the typings from the paper checkbox and using rest parameters to pass everything at once to the checkbox? (Whilst making sure to omit certain things we have to explicitly set)

File: src/components/checkBox.tsx [hunk 1]
```diff
@@ -30,11 +40,6 @@ const styles = StyleSheet.create({
     flexDirection: 'row',
     alignItems: 'center',
     flex: 1,
-    marginBottom: 1.5,
```
Comment (Thodor12) [inline src/components/checkBox.tsx:33]: Removal of this margin isn't going to cause items butted up too close together?

**Review by Thodor12 — COMMENTED** [review]
Could I also get a before and after of how it looks, with pure basic configuration, no modified theming, etc.
If the UI changes it might have to be marked as a breaking change.

File: src/components/checkBox.tsx (hunk 1 above)
Comment (himrocks33) [reply src/components/checkBox.tsx:33]: No there is plenty of padding on Checkbox.Item already no need for additional.

**Comment by himrocks33** [comment]
@Thodor12 UI changes are minimal.   Seems like you are grilling me for no reason on this PR.  It’s pretty straightforward.

**Comment by srivastavaanurag79** [comment]
**@himrocks33**  Please add this feature to this merge request,i.e, limiter, a key name `limit` which limits the maximum number of items one can select in the multi-select.

**Future Features**:
1. Sectional List i.e.  see below example

- [ ] ASIA
  - [ ] UAE
  - [ ] INDIA
  - [ ] NEPAL
- [ ] EUROPE
  - [ ] UK
  - [ ] SPAIN
  - [ ] FRANCE

2. Chips style selected items (optional styling)
3. Different View For Single Select, example below
1_rQNkcjLw1RYtDZRNqoyQtw

**Comment by Thodor12** [comment]
> @Thodor12 UI changes are minimal. Seems like you are grilling me for no reason on this PR. It’s pretty straightforward.

If you think that that is your problem.

All I'm asking for is a simple before and after. I see a bunch of changes and I've got no idea of the impact of this.
If things start changing between versions we have to state it as a breaking change, otherwise people can get silently a new interface without knowing if they update the library.

**Comment by srivastavaanurag79** [comment]
It will be a breaking change since the checkbox style is being changed and some other new features are being added as well. The pull request will be strictly reviewed so please be patient and please be a bit polite everyone.

**Comment by srivastavaanurag79** [comment]
@Thodor12 If you have time please switch to @himrocks33 branch and run the project to review the changes and update it with your guidance if you find something unusual or not needed.

**Comment by himrocks33** [comment]
@srivastavaanurag79 #59 items seem like a complete rewrite almost to support the single select (which would probably be a whole new component really?)  I doubt I have time for that right now.  Bug Fixes and minor changes I can do.  Where do we stand with getting this PR merged?

**Comment by srivastavaanurag79** [comment]
@himrocks33 #59 you Don't have to do them, those are future features

**Comment by srivastavaanurag79** [comment]
@himrocks33 if adding a limiter feature is possible please add it, a key name limit which limits the maximum number of items one can select in the multi-select, will try to review and merge this pull request this Sunday by myself if other contributors don't have time for the same.

**Comment by srivastavaanurag79** [comment]
Fixed disable issue and checkbox label issue, select all issue
Screenshot 2024-04-18 at 2 53 21 PM

**Review by srivastavaanurag79 — COMMENTED** [review]
I have reviewed all the changes. The major change was in the checkbox. Touchable Opacity was removed but it would not break any existing installed version. The code looks fine. I addressed the issues which I found.

**Pull request merged**